lib/internal/util/types.js

```diff
@@ -2,10 +2,18 @@
 
 const {
   ArrayBufferIsView,
+  Object,
   ObjectDefineProperties,
   TypedArrayPrototypeGetSymbolToStringTag,
 } = primordials;
 
+function isPlainObject(value) {
+  if (typeof value !== 'object') return false;
+  if (value.constructor && value.constructor !== Object) return false;
+
+  return true;
+}
```

Comment:
@aduh95 Do we not have a check for this already in the codebase somewhere?

@JakobJingleheimer Where did you get this from? We should add a citation in a comment. In particular, why this logic and not what’s in https://github.com/sindresorhus/is-plain-obj/blob/main/index.js?

Reply:
This implementation will throw an error if `value === null`.

Reply:
I wrote it because I couldn't find an existing one in the codebase. Happy to swap it out for something else.